src/app/component/MyAccount/MyAccount.component.js

```diff
@@ -0,0 +1,609 @@
+                confirmpassword: '',
+                addressfirstname: customerData.addressfirstname || customerData.firstname,
+                addresslastname: customerData.addresslastname || customerData.lastname
+            }
+        });
+        this.button.current.focus();
+    }
+
+    /**
+     * Go back to login page when clicking on account button
+     */
+    goBackToDefault() {
+        const { state, isOpen } = this.state;
+
+        if (state !== STATE_LOGGED_IN && state !== STATE_VALIDATE_SIGN_UP) {
+            this.setState({ state: STATE_SIGN_IN, fieldsToValidate: [] });
+        }
+
+        this.setState({ isOpen: !isOpen });
+    }
+
+    /**
+     * Render Button
+     */
+    renderButton() {
+        const { state, isOpen } = this.state;
+        const actionText = state === STATE_LOGGED_IN
+            ? 'Hello, User'
+            : 'My Account';
+
+        return (
+            <button
+              block="MyAccount"
+              elem="Button"
+              ref={ this.button }
+              mods={ { isOpen } }
+              onClick={ () => this.goBackToDefault() }
+            >
+                <i
+                  block="MyAccount"
+                  elem="Icon"
+                />
+                <span>{ actionText }</span>
+            </button>
+        );
+    }
+
+    /**
+     * Render Dropdown
+     */
+    renderDropdown() {
+        const { state } = this.state;
+        const renderFunction = this.renderMap[state];
+
+        return (
+            <div block="MyAccount" elem="Dropdown" mods={ { state } }>
+                <div block="MyAccount" elem="Action" mods={ { state } }>
+                    { renderFunction() }
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Render Account Actions
+     */
+    renderAccountActions() {
+        return (
+            <nav block="MyAccount" elem="Navigation">
+                <ul>
+                    <li><a>My Account</a></li>
+                    <li><a>My Orders</a></li>
+                    <li><a>Logout</a></li>
+                </ul>
+            </nav>
+        );
+    }
+
+    /**
+     * Render Forgot Password page
+     */
+    renderForgotPassword() {
+        const { fieldsToValidate, customerData } = this.state;
+
+        return (
+            <>
+                <form>
+                    <h3>Get password reset link</h3>
+                    <Field
+                      type="text"
+                      label="Email"
+                      id="email"
+                      key="email"
+                      handleToUpdate={ this.handleToUpdate }
+                      originalValue={ customerData.email }
+                    />
+                    <div block="MyAccount" elem="Buttons">
+                        <button disabled={ fieldsToValidate.length !== 0 }>Send reset link</button>
+                    </div>
+                </form>
+                <article block="MyAccount" elem="Additional">
```

Comment:
I think its wrong application of article tag.  https://html.com/tags/article/
Why you have decided to use it?